beacon-chain/sync/simulated_sync_test.go

```diff
@@ -0,0 +1,233 @@
+	if !ok {
+		return
+	}
+
+	feed.Send(p2p.Message{Ctx: sim.ctx, Data: msg})
+}
+
+func setupSimBackendAndDB(t *testing.T) (*backend.SimulatedBackend, *db.BeaconDB, []*bls.SecretKey) {
+	bd, err := backend.NewSimulatedBackend()
+	if err != nil {
+		t.Fatalf("Could not set up simulated backend %v", err)
+	}
+
+	privKeys, err := bd.SetupBackend(100)
+	if err != nil {
+		t.Fatalf("Could not set up backend %v", err)
+	}
+
+	beacondb, err := db.SetupDB()
+	if err != nil {
+		t.Fatalf("Could not setup beacon db %v", err)
+	}
+
+	if err := beacondb.SaveState(bd.State()); err != nil {
+		t.Fatalf("Could not save state %v", err)
+	}
+
+	memBlocks := bd.InMemoryBlocks()
+	if err := beacondb.SaveBlock(memBlocks[0]); err != nil {
+		t.Fatalf("Could not save block %v", err)
+	}
+
+	if err := beacondb.UpdateChainHead(memBlocks[0], bd.State()); err != nil {
+		t.Fatalf("Could not update chain head %v", err)
+	}
+
+	return bd, beacondb, privKeys
+}
+
+func setUpSyncedService(numOfBlocks int, simP2P *simulatedP2P, t *testing.T) (*Service, *db.BeaconDB) {
+	bd, beacondb, privKeys := setupSimBackendAndDB(t)
+	defer bd.Shutdown()
+	defer db.TeardownDB(bd.DB())
+
+	mockPow := &genesisPowChain{
+		feed: new(event.Feed),
+	}
+
+	mockChain := &mockChainService{
+		feed: new(event.Feed),
+	}
+
+	cfg := &Config{
+		ChainService:     mockChain,
+		BeaconDB:         beacondb,
+		OperationService: &mockOperationService{},
+		P2P:              simP2P,
+		PowChainService:  mockPow,
+	}
+
+	ss := NewSyncService(context.Background(), cfg)
+
+	go ss.run()
+	for !ss.Querier.chainStarted {
+		mockPow.feed.Send(time.Now())
+	}
+
+	for i := 1; i <= numOfBlocks; i++ {
+		if err := bd.GenerateBlockAndAdvanceChain(&backend.SimulatedObjects{}, privKeys); err != nil {
+			t.Fatalf("Unable to generate block in simulated backend %v", err)
+		}
+		blocks := bd.InMemoryBlocks()
+		if err := beacondb.SaveBlock(blocks[i]); err != nil {
+			t.Fatalf("Unable to save block %v", err)
+		}
+		if err := beacondb.UpdateChainHead(blocks[i], bd.State()); err != nil {
+			t.Fatalf("Unable to update chain head %v", err)
+		}
+	}
+
+	return ss, beacondb
+}
+
+func setUpUnSyncedService(simP2P *simulatedP2P, t *testing.T) (*Service, *db.BeaconDB) {
+	bd, beacondb, _ := setupSimBackendAndDB(t)
+	defer bd.Shutdown()
+	defer db.TeardownDB(bd.DB())
+
+	mockPow := &afterGenesisPowChain{
+		feed: new(event.Feed),
+	}
+
+	mockChain := &mockChainService{
+		feed: new(event.Feed),
+	}
+
+	cfg := &Config{
+		ChainService:     mockChain,
+		BeaconDB:         beacondb,
+		OperationService: &mockOperationService{},
+		P2P:              simP2P,
+		PowChainService:  mockPow,
+	}
+
+	ss := NewSyncService(context.Background(), cfg)
+
+	go ss.run()
+
+	for ss.Querier.curentHeadSlot == 0 {
+		simP2P.Send(&pb.ChainHeadResponse{
+			Slot: params.BeaconConfig().GenesisSlot + 10,
+			Hash: []byte{'t', 'e', 's', 't'},
+		}, p2p.Peer{})
+	}
+
+	return ss, beacondb
+}
+
+func TestSyncFromAFullySyncedNode(t *testing.T) {
```

Comment:
Follow the underscore pattern for test names here

Reply:
Done